ctapipe/calib/camera/flatfield.py

```diff
@@ -0,0 +1,247 @@
+
+from ctapipe.image import ChargeExtractorFactory, WaveformCleanerFactory
+from ctapipe.core.traits import Int
+from ctapipe.io.containers import FlatFieldCameraContainer
+
+__all__ = [
+    'FlatFieldCalculator',
+    'FlasherFlatFieldCalculator',
+    'FlatFieldFactory'
+]
+
+
+class FlatFieldCalculator(Component):
+    """
+    Parent class for the flat field calculators. Fills the MON.flatfield container.
+
+    """
+    max_time_range_s = Int(60, help='Define the maximum time interval per'
+           ' coefficient flat-filed calculation').tag(config=True)
+    max_events = Int(10000, help='Define the maximum number of events per '
+           ' coefficient flat-filed calculation').tag(config=True)
+    n_channels = Int(2, help='Define the number of channels to be '
+                    'treated ').tag(config=True)
+
+    def __init__(self, config=None, tool=None, extractor_product=None, cleaner_product=None, **kwargs):
+        """
+        Parent class for the flat field calculators. Fills the MON.flatfield container.
+
+        Parameters
+        ----------
+        config : traitlets.loader.Config
+            Configuration specified by config file or cmdline arguments.
+            Used to set traitlet values.
+            Set to None if no configuration to pass.
+        tool : ctapipe.core.Tool
+            Tool executable that is calling this component.
+            Passes the correct logger to the component.
+            Set to None if no Tool to pass.
+        extractor_product : str
+            The ChargeExtractor to use.
+        cleaner_product : str
+            The WaveformCleaner to use.
+        kwargs
+
+        """
+        super().__init__(config=config, parent=tool, **kwargs)
+
+        # initialize the output
+        self.container = FlatFieldCameraContainer()
+
+        # load the waveform charge extractor and cleaner
+        kwargs_ = dict()
+        if extractor_product:
+            kwargs_['product'] = extractor_product
+        self.extractor = ChargeExtractorFactory.produce(
+            config=config,
+            tool=tool,
+            **kwargs_
+        )
+        self.log.info(f"extractor {self.extractor}")
+
+        kwargs_ = dict()
+        if cleaner_product:
+            kwargs_['product'] = cleaner_product
+        self.cleaner = WaveformCleanerFactory.produce(
+            config=config,
+            tool=tool,
+            **kwargs_
+        )
+        self.log.info(f"cleaner {self.cleaner}")
+
+    @abstractmethod
+    def calculate_relative_gain(self,event):
+        """
+        Parameters
+        ----------
+        event
+
+        """
+
+
+class FlasherFlatFieldCalculator(FlatFieldCalculator):
+    """
+    Class for calculating flat field coefficients witht the
+    flasher data. Fills the MON.flatfield container.
+    """
+
+    def __init__(self, config=None, tool=None, **kwargs):
+        """
+        Parent class for the flat-field calculators.
```

Comment:
This is lying documentation as a result of a copy+paste action I assume. We as developers must be very careful not to fall into copy & paste traps, because there is a rule, that documentation must be written. 

Look at this particular case: the `config` and `tool` parameters are found all over our code base. Does it really add value, if we copy and paste their description into every Tool and Component we have? 

If we remove the description of `config` and `tool` what is left?
```
Parent class for the flat-field calculators.
Fills the MON.flatfield container.
kwargs
```

Of which the 1st line is a lie. The 2nd is already contained in the class-doc-string. The 3rd line is actually interesting, as it could explain what are the other parameters for this Component... but unfortunately it was omitted.
I assume a doc-string which is visually already very long is assumed to be complete ... another trap which we should try to avoid, I believe.